src/field/field3d.cxx

```diff
@@ -198,16 +194,23 @@ void Field3D::mergeYupYdown() {
   if(yup_field == this && ydown_field == this)
     return;
 
-  if(yup_field != nullptr){
+  clearYupYdown();
+
+  yup_field = this;
+  ydown_field = this;
+}
+
+void Field3D::clearYupYdown() {
+  // Delete auxiliary fields if they have been set
+  if (yup_field != nullptr && yup_field != this) {
     delete yup_field;
   }
+  yup_field = nullptr;
 
-  if(ydown_field != nullptr) {
+  if (ydown_field != nullptr && ydown_field != this) {
     delete ydown_field;
   }
-
-  yup_field = this;
-  ydown_field = this;
+  ydown_field = nullptr;
```

Comment:
A fractionally better way of writing this:

```cpp
void Field3D::clearYupYdown() {
  if (yup_field == this && ydown_field == this) {
    return;	
  }

  delete yup_field;
  yup_field = nullptr;

  delete ydown_field;
  ydown_field = nullptr;
}
```

1. Preconditions/assumptions up-front with early exit
2. No guard against deleting `nullptr` (this is fine in C++)